qiskit/transpiler/passes/utils/filter_op_nodes.py

```diff
@@ -0,0 +1,63 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Remove all babeled ops from a circuit"""
+
+from typing import Callable
+
+from qiskit.dagcircuit import DAGCircuit, DAGOpNode
+from qiskit.transpiler.basepasses import TransformationPass
+from qiskit.transpiler.passes.utils import control_flow
+
+
+class FilterOpNodes(TransformationPass):
+    """Remove all operations that match a filter function
+
+    This transformation pass is used to remove any operations that matches a
+    the provided filter function.
+
+    Args:
+       predicate: A given callable that will be passed the :class:`.DAGOpNode`
+           for each node in the :class:`.DAGCircuit`.
```

Comment:
Relatively minor usability note: the polarity of the predicate here is backwards to the "standard" functional-programming `filter` primitive, including Python's built-in.  Do we potentially want to match the built-in (so the return value corresponds to "retain", as opposed to the current structure of "remove"), or do you think that'll lead to more awkward predicates?

Reply:
I'm fine with inverting it, this is what felt natural to me but consistency with others expectations is important. I am the weird guy who constantly writes code like:

```python
if not condition:
     pass
else:
     pass
```

so my inclinations are probably a bit off.

Reply:
I updated this in: https://github.com/Qiskit/qiskit/pull/10323/commits/f6df5be6021c7acd9c28151f9278631c7d203e78 and also clearly documented the behavior.